.github/workflows/build.yml

```diff
@@ -0,0 +1,266 @@
+      - name: Configure
+        run: |
+          mkdir build && cd build
+          ../cmake/build_win${{ matrix.arch }}.sh
+      - name: Build
+        run: cmake --build build
+      - name: Build tests
+        run: cmake --build build --target tests
+      - name: Package
+        run: cmake --build build --target package
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: mingw${{ matrix.arch }}
+          path: build/lmms-*.exe
+      - name: Print ccache statistics
+        run: |
+          echo "[ccache config]"
+          ccache -p
+          echo "[ccache stats]"
+          ccache -s
+  msvc:
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: ['x86', 'x64']
+    name: msvc-${{ matrix.arch }}
+    runs-on: windows-2019
+    env:
+      qt-version: '5.15.2'
+    steps:
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          submodules: recursive
+      - name: Cache Qt
+        id: cache-qt
+        uses: actions/cache@v3
+        with:
+          key: qt-${{ env.qt-version }}-msvc-${{ matrix.arch }}-0
+          path: Qt
+      - name: Cache vcpkg dependencies
+        uses: actions/cache@v3
+        with:
+          key: vcpkg-${{ matrix.arch }}-${{ github.ref }}-${{ github.run_id }}
+          restore-keys: |
+            vcpkg-${{ matrix.arch }}-${{ github.ref }}-
+            vcpkg-${{ matrix.arch }}-
+          path: C:\vcpkg\installed
+      - name: Install Qt
+        run: |
+          $qt_version = ${env:qt-version}
+          $target_dir = Join-Path (Get-Location) 'Qt'
+          if ('${{ steps.cache-qt.outputs.cache-hit }}' -ne 'true') {
+            pip install aqtinstall
+            switch ('${{ matrix.arch }}') {
+              'x86' {
+                aqt install-qt windows desktop $qt_version win32_msvc2019 `
+                  --archives qtbase qtsvg qttools -O $target_dir
+              }
+              'x64' {
+                aqt install-qt windows desktop $qt_version win64_msvc2019_64 `
+                  --archives qtbase qtsvg qttools -O $target_dir
+                aqt install-qt windows desktop $qt_version win32_msvc2019 `
+                  --archives qtbase -O $target_dir
+              }
+            }
+          }
+          $qt_arch = switch ('${{ matrix.arch }}') {
+            'x86' { 'msvc2019' }
+            'x64' { 'msvc2019_64' }
+          }
+          $qt_dir = Join-Path $target_dir $qt_version $qt_arch
+          $qt_bin_dir = Join-Path $qt_dir 'bin'
+          echo "Qt5_DIR=$qt_dir" >> $env:GITHUB_ENV
+          echo $qt_bin_dir >> $env:GITHUB_PATH
+      - name: Install dependencies
+        run: |
+          vcpkg install --triplet ${{ matrix.arch }}-windows --recurse `
+            fftw3 libsamplerate libsndfile lilv lv2 sdl2
+      - name: Set up build environment
+        uses: ilammy/msvc-dev-cmd@d8610e2b41c6d0f0c3b4c46dad8df0fd826c68e1
```

Comment:
Why commit hash instead of tag name?

Reply:
GitHub recommend hashes for security reasons (you can't point them somewhere else without finding a collision, unlike tags): https://docs.github.com/en/actions/security-guides/security-hardening-for-github-actions#using-third-party-actions

I don't think it's that likely to be an issue though, so I'm happy to switch to a tag if you think that would be better.

Reply:
That makes sense.